service/src/main/java/bio/terra/workspace/app/controller/ControlledFlexibleResourceApiController.java

```diff
@@ -210,61 +209,50 @@ public ResponseEntity<ApiCloneControlledFlexibleResourceResult> cloneFlexibleRes
+        resource.computeCloningInstructions(body.getCloningInstructions());
+    ResourceValidationUtils.validateCloningInstructions(
+        StewardshipType.CONTROLLED, cloningInstructions);
     workspaceService.validateWorkspaceState(workspaceUuid);
-    workspaceService.validateWorkspaceState(body.getDestinationWorkspaceId());
-
-    if (body.getCloningInstructions() != null) {
-      ResourceValidationUtils.validateCloningInstructions(
-          StewardshipType.CONTROLLED,
-          CloningInstructions.fromApiModel(body.getCloningInstructions()));
-    }
-
-    ControlledResource sourceFlexResource =
-        controlledResourceService.getControlledResource(workspaceUuid, resourceUuid);
-
-    CloningInstructions effectiveCloningInstructions =
-        Optional.ofNullable(body.getCloningInstructions())
-            .map(CloningInstructions::fromApiModel)
-            .orElse(sourceFlexResource.getCloningInstructions());
-
-    UUID sourceResourceId = sourceFlexResource.getResourceId();
-    UUID sourceWorkspaceId = sourceFlexResource.getWorkspaceId();
+    workspaceService.validateCloneWorkspaceAndContextState(
+        body.getDestinationWorkspaceId(), CloudPlatform.ANY, cloningInstructions);
 
     // If COPY_NOTHING return an empty result (no-op).
-    if (effectiveCloningInstructions == CloningInstructions.COPY_NOTHING) {
+    if (cloningInstructions == CloningInstructions.COPY_NOTHING) {
       ApiCloneControlledFlexibleResourceResult emptyResult =
           new ApiCloneControlledFlexibleResourceResult()
               .effectiveCloningInstructions(CloningInstructions.COPY_NOTHING.toApiModel())
-              .sourceResourceId(sourceResourceId)
-              .sourceWorkspaceId(sourceWorkspaceId)
+              .sourceResourceId(resourceUuid)
+              .sourceWorkspaceId(workspaceUuid)
               .resource(null);
       return new ResponseEntity<>(emptyResult, HttpStatus.OK);
     }
 
     // Otherwise start a flight to clone the flex resource.
     ControlledFlexibleResource clonedFlexResource =
         controlledResourceService.cloneFlexResource(
-            workspaceUuid,
-            resourceUuid,
+            resource,
             body.getDestinationWorkspaceId(),
             UUID.randomUUID(),
             userRequest,
             body.getName(),
             body.getDescription(),
-            body.getCloningInstructions());
+            cloningInstructions);
 
     ApiCloneControlledFlexibleResourceResult result =
         new ApiCloneControlledFlexibleResourceResult()
             .resource(clonedFlexResource.toApiResource())
-            .effectiveCloningInstructions(effectiveCloningInstructions.toApiModel())
-            .sourceWorkspaceId(sourceResourceId)
-            .sourceResourceId(sourceWorkspaceId);
+            .effectiveCloningInstructions(cloningInstructions.toApiModel())
+            .sourceWorkspaceId(resourceUuid)
+            .sourceResourceId(workspaceUuid);
```

Comment:
you didn't introduce this, but it looks like the resource/workspace ID arguments are swapped here

Reply:
Ha! And since it is setting the return, it doesn't break anything.
Good eyes! 👀 
I fixed it.